### DISCUSSION TITLE
SJTUThesis v2 Roadmap

### DISCUSSION THREAD

**Discussion by skyzh** [sjtug/SJTUThesis#705] [灌水区]
* 合并 SJTUThesis 和 SJTUTeX https://github.com/sjtug/SJTUThesis/discussions/630
* 研究生模版和本科生模版差异较大，看起来需要一个单独的 dtx https://github.com/sjtug/SJTUThesis/issues/652
* 和教务/图书馆核对本科生、研究生模版的格式是否符合要求
* 添加课程论文模版（感觉 SJTUBeamer 的 user guide 模版挺好看的？）
* 更新文档 && 添加单元测试 （比如 https://github.com/sjtug/SJTUThesis/issues/694 看起来就是文档没有及时更新的例子）

各位还有啥 idea 吗（

cc @AlexaraWu @LogCreative

**Comment by skyzh** [comment]
关于 SJTUTeX 和 SJTUThesis 的合并，我觉得其实可以把 SJTUTeX 作为 SJTUThesis 的 git submodule，这样就不用一直互相同步了……

**Comment by AlexaraWu** [comment]
用 LaTeX3 重构？

写了一个 Prototype，拆了四个子类。

![image](https://user-images.githubusercontent.com/15205102/132126347-d30e01c6-afb6-4fc2-b22d-e2e75a6e27e5.png)

**Reply by skyzh** [reply]
做梦都想！看起来 work 🤩

**Reply by skyzh** [reply]
四个不同的 cls 都是用 dtx 生成的？要不要先开个 develop 分支（

**Reply by AlexaraWu** [reply]
是一个 dtx 生成的。放一个仓库里吗？感觉代码没有延续性（

**Reply by skyzh** [reply]
反正都要升级 2.0 了，我们可以在 develop 上从头开始造，最后 merge 到主分支上（

**Reply by AlexaraWu** [reply]
也可以。

**Reply by AlexaraWu** [reply]
单放了一个 dtx（

**Reply by LogCreative** [reply]
l3 厉害了

**Reply by AlexaraWu** [reply]
重开了独立的 v2 分支，rebase 到 root 了。开发完成后将 v2 设为主分支，不再 merge 到原 master 上，原 master 之后改为 v1 归档。

**Reply by skyzh** [reply]
我更倾向于保留历史记录，到时候把 master 和 develop 换一下（

**Reply by skyzh** [reply]
通常来讲项目的大重构都是保留历史的（

**Reply by AlexaraWu** [reply]
v1 (master) 分支在，历史记录就在呀

**Reply by skyzh** [reply]
> v1 (master) 分支在，历史记录就在呀

GitHub 项目主页的 contributors 默认只显示主分支上的贡献者（

**Reply by AlexaraWu** [reply]
> GitHub 项目主页的 contributors 默认只显示主分支上的贡献者（

唔……确实

那 v2 开发还是移到 SJTUTeX 做好了，之后在这里更新 texmf 里的内容。

主要是课程论文的模板我想基于 ctexart 单独做一个，这里放一起不是很合适。

**Reply by AlexaraWu** [reply]
article 一级标题是 section，不方便共用示例文档。

**Reply by skyzh** [reply]
关于要不要分开，我倒是觉得放一起好，主要有几点考虑：
* 改模版的时候可能 somehow 会导致示例模版编不过，放一起可以早发现这个问题。
* 发布的时候也可以把示例模版和 sjtuthesis 放一起发布，用户可以在一个地方找到示例文档和开发文档 (FYI https://github.com/sjtug/SJTUBeamer/releases/tag/v2.0.0)
* 两边同步起来会比较麻烦（

关于可能的一些 concern：
* 关于代码包体积，基本上都是文本文件，大不了多少
* 关于用户使用，现在 SJTUBeamer 的做法是开发者修改 dtx 后要更新根目录下的 sty 文件，保证任何时候 master 上的代码既可以通过 l3build 编译开发文档，用户也可以用 latexmk 编译根目录下的示例文档
* 关于共享示例文档，我觉得内容上基本上都一致，无非就是 \section 还是 \chapter。我们完全可以把每一章/节的内容写进 tex 文件里，然后同时提供 course 和 graduate 的示例文档。示例文档的内容就是定义一些 section / chapter 然后把内容引入进来（

**Reply by skyzh** [reply]
我先把 develop 分支恢复一下？

**Reply by skyzh** [reply]
其实我比较好奇你是怎么开发的 🤣🤣 比如改完模版以后怎么看效果，我感觉我也是最近才熟悉 dtx 的 workflow，之前想改 SJTUTeX 都无从下手（逃

**Reply by LogCreative** [reply]
俺也好奇

**Reply by AlexaraWu** [reply]
> 其实我比较好奇你是怎么开发的 🤣🤣 比如改完模版以后怎么看效果，我感觉我也是最近才熟悉 dtx 的 workflow，之前想改 SJTUTeX 都无从下手（逃

`.latexmkrc` 的 `TEXINPUTS` 路径改一下不就可以了？

**Reply by skyzh** [reply]
对，但每次看效果都要先跑 l3build，再跑 latexmk。而且对于其他未来的社区开发者来讲，分两个仓库还增加了他们设置环境的开销 😛

现在所有东西塞进一个 repo 以后，普通用户跑 latexmk 或者 make thesis 就行了；想改模版的用户 make build-dev 就能一键编译 dtx 和示例文档（
